utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -581,16 +581,23 @@ bool WritePreparedTxnDB::IsInSnapshot(uint64_t prep_seq,
   return false;
 }
 
-void WritePreparedTxnDB::AddPrepared(uint64_t seq) { prepared_txns_.push(seq); }
+void WritePreparedTxnDB::AddPrepared(uint64_t seq) {
+  WriteLock wl(&prepared_mutex_);
+  prepared_txns_.push(seq);
+  ROCKS_LOG_INFO(db_impl_->GetDBOptions().info_log, "Txn %lu Prepared", seq);
```

Comment:
better cache info_log. DBImpl::GetDBOptions() return a copy of the options, not a reference.